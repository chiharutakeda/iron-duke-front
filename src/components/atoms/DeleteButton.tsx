import { gql } from 'graphql-request';
import { gqlfetcher } from '@/fetch/fetcher';
import { DeleteToDOVariables } from '@/generated/DeleteToDO';

//この変数に与える型はサーバー側の型をschemaを参照する。
const deleteToDolist_M = gql`
  mutation DeleteToDO($deleteTODO: TodoListDeleteInputType!) {
    DeleteToDo(deleteTODO: $deleteTODO) {
      id
      firstName
      lastName
    }
  }
`;

const style = {
  margin: '3px 0px',
  height: '2rem',
  width: 'auto',
  color: 'black',
  background: '#CC0000',
  borderRadius: '5px',
  borderStyle: 'none',
  boxShadow: '1px 1px 4px',
  cursor: 'pointer',
};

const DeleteButton = (deleteId: { ID: number }) => {
  return (
    <button
      style={style}
      onClick={() => {
        const deleteToDolist_V: DeleteToDOVariables = { deleteTODO: deleteId };
        gqlfetcher(deleteToDolist_M, deleteToDolist_V);
      }}
    >
      Delete
    </button>
  );
};

export default DeleteButton;
