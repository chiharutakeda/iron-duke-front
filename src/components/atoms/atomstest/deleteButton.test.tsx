import ReactTestRenderer from 'react-test-renderer';
import DeleteButton from '../DeleteButton';

it('DeleteButton', async () => {
  const tree = await ReactTestRenderer.create(
    <DeleteButton ID={1} />
  ).toJSON();
  await expect(tree).toMatchSnapshot();
});
