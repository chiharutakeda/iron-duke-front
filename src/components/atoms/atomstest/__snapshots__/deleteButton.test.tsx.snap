// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteButton 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "background": "#CC0000",
      "borderRadius": "5px",
      "borderStyle": "none",
      "boxShadow": "1px 1px 4px",
      "color": "black",
      "cursor": "pointer",
      "height": "2rem",
      "margin": "3px 0px",
      "width": "auto",
    }
  }
>
  Delete
</button>
`;
